File: CS/ZoomPictureEdit/ZoomPictureEditViewInfo.cs
using System;
using System.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace nsZoomPictureEdit
{
    public class ZoomPictureEditViewInfo : PictureEditViewInfo
    {
        public ZoomPictureEditViewInfo(DevExpress.XtraEditors.Repository.RepositoryItem item)
            : base(item)
        {
        }

        protected override void CalcImageRect(Rectangle bounds)
        {
            float zoomFactor = ((RepositoryItemZoomPictureEdit)Item).fZoomFactor;

            Rectangle originRect = bounds;
            bounds.Width = (int)Math.Round(bounds.Width * zoomFactor);
            int horizIndent = (originRect.Width - bounds.Width) / 2;
            bounds.X += horizIndent;

            bounds.Height = (int)Math.Round(bounds.Height * zoomFactor);
            int vertIndent = (originRect.Height - bounds.Height) / 2;
            bounds.Y += vertIndent;

            base.CalcImageRect(bounds);
        }
    }
}

File: VB/ZoomPictureEdit/ZoomPictureEditViewInfo.vb
Imports System
Imports System.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports Microsoft.VisualBasic

Namespace nsZoomPictureEdit
	Public Class ZoomPictureEditViewInfo
		Inherits PictureEditViewInfo
		Public Sub New(ByVal item As DevExpress.XtraEditors.Repository.RepositoryItem)
			MyBase.New(item)
		End Sub

		Protected Overrides Sub CalcImageRect(ByVal bounds As Rectangle)
			Dim zoomFactor As Single = (CType(Item, RepositoryItemZoomPictureEdit)).fZoomFactor

			Dim originRect As Rectangle = bounds
			bounds.Width = CInt(Fix(Math.Round(bounds.Width * zoomFactor)))
            Dim horizIndent As Integer = (originRect.Width - bounds.Width) / 2
			bounds.X += horizIndent

			bounds.Height = CInt(Fix(Math.Round(bounds.Height * zoomFactor)))
            Dim vertIndent As Integer = (originRect.Height - bounds.Height) / 2
			bounds.Y += vertIndent

			MyBase.CalcImageRect(bounds)
		End Sub
	End Class
End Namespace
